### ISSUE TITLE
Module version declarations have circular dependency on this distribution

### ISSUE THREAD

**Issue by Grinnz** [evalEmpire/perl5i#303]
The line where a module version is assigned is parsed by static parsers, so it needs to be fully executable on its own. Many of the modules in this distribution declare their version like so:

    use perl5i::VERSION; our $VERSION = perl5i::VERSION->VERSION;

This is a problem. In order for this to work for anyone without this distribution installed yet, it would need to declare perl5i::VERSION as a dependency. However, perl5i::VERSION is in this distribution, and declaring a dependency on yourself obviously doesn't work. Please replace this with a real version assignment and maintain your versions using a tool like https://metacpan.org/pod/App::RewriteVersion.

**Comment by schwern** [comment]
@Grinnz I see, you're right. App::RewriteVersion looks pretty nice, I'd be happy to see a patch to use it.

**Comment by Grinnz** [comment]
It wouldn't be something the tooling could use itself (unless using something like dzil with the related RewriteVersion plugin). It was just a suggestion for what could be used to mass-update versions as an author. But I could provide a patch to fix the current version assignments.

**Comment by schwern** [comment]
Yes, a better fix for how we do version assignments internally would be helpful. The current version is a bit hacky. Something to replace the change_version script https://github.com/evalEmpire/perl5i/blob/master/admin/change_version and make the `$VERSION` a simple number would be great.
